awx/main/middleware.py

```diff
@@ -199,9 +201,21 @@ def process_request(self, request):
             request.path_info = new_path
 
 
+@memoize(ttl=20)
+def is_migrating():
+    last_applied = MigrationRecorder(connection).migration_qs.order_by('-applied').only('name').first().name
```

Comment:
This is a stripped-down query compared to what we had before, which did _multiple_ queries to return _the full list_ of migrations... which grows over time.

But I think it should be stripped down even more. Probably want to reverse the order, and find the largest migration number from the module, then do a `.exists()` query.

Also think I had it locally wrong, where we need to `.filter(applied__isnull=False)`, so that we only count migrations that have been applied. We might also arguably add `app="main"` to the filter.